src/App.tsx

```diff
@@ -15,48 +16,99 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  SORT_FIELD_ALPH = 'alph',
+  SORT_FIELD_LENGTH = 'length',
+  SORT_FIELD_NONE = '',
+}
+
+function getPreparedGoods(
+  goods: string[],
+  sortField: SortType,
+  toReverse: boolean,
+) {
+  const preparedGoods = [...goods];
+
+  if (sortField) {
+    preparedGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case SortType.SORT_FIELD_ALPH:
+          return good1.localeCompare(good2);
+
+        case SortType.SORT_FIELD_LENGTH:
+          return good1.length - good2.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (toReverse) {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState(SortType.SORT_FIELD_NONE);
+  const [reversedField, setReversedField] = useState(false);
```

Comment:
It is better to use more descriptive names for your state variables. For example, 'isReversed' instead of 'reversedField'.